shell/platform/darwin/ios/framework/Source/SemanticsObject.mm

```diff
@@ -174,7 +176,9 @@ - (void)accessibilityBridgeDidFinishUpdate {
   // contentOffset is 0.0, only the scroll down action is available.
   self.scrollView.frame = self.accessibilityFrame;
   self.scrollView.contentSize = [self contentSizeInternal];
-  [self.scrollView setContentOffset:[self contentOffsetInternal] animated:NO];
+  if (!self.scrollView.isDoingSystemScrolling) {
```

Comment:
This means if the scroll view is in a FKA scrolling animation, all framework updates will be disregarded. 

This is not optimal, if the user wants to interrupt the scrolling animation via swipe gestures, the `UIScrollView` will be temporarily out of sync with the framework scroll offset, but once the animation is done, this `UIScrollView` will send one last offset update to the framework, so it's guaranteed that the framework semantics tree will need another update and the new tree will also update the accessibility bridge and the offsets maintained by the scroll view and the framework will become consistent again.

Reply:
I have tried removing this flag, and the offsets echoed back by the framework seem always to be one step behind and I am not sure exactly why:

It seems `Shell::OnPlatformViewDispatchSemanticsAction` and `Shell::OnEngineUpdateSemantics` both run tasks using `RunNowOrPostTask`. Now that the UI thread is the same as the platform thread, there should be no delay, I was expecting:
```
<<<FKA setContentOffset: {0, 13.333333333333334}
<<<FKA setContentOffset: {0, 13.333333333333334}
>>>framework setContentOffset: {0, 13.333333333333343}
<<<FKA setContentOffset: {0, 15}
<<<FKA setContentOffset: {0, 15}
>>>framework setContentOffset: {0, 15}
```

but instead it's always delayed:
```
<<<FKA setContentOffset: {0, 13.333333333333334}
<<<FKA setContentOffset: {0, 13.333333333333334}
... 
<<<FKA setContentOffset: {0, 15}
<<<FKA setContentOffset: {0, 15}
>>>framework setContentOffset: {0, 15}
```

Reply:
yes, i think this makes sense since FKA set content will change the scrollcontroller, but it will have to wait for one frame to get the layout and semantics update back

Reply:
Please add a comment covering this here.

Reply:
So it looks like I was looking at an older checkout, and it was changed [here](https://github.com/flutter/engine/pull/56514). The event ordering becomes the way I expected it to be if we run `DispatchSemanticsAction` synchronously like before. Jonah said `RunNowOrPostTask` semantics action dispatch is fine we'll just need to post an empty task afterwards to flush the dart futures. I think if we do that instead we could eliminate this flag.

That seems to be the better option because we don't need the flag and the scroll view will be responsive to user gestures during the FKA scroll animation, but I think I'll make the attempt a separate PR.

Reply:
> I am curious to see if it will work, but I think even if the action is send to framework synchronously, it may still take one frame for flutter to rebuild layout and flush semantics back

That's fine I think. The problem with `DispatchSemanticsAction` being async is that we are essentially maintaining two sources of truth so it becomes a distributed system where the communication channels only guarantee FIFO ordering. If the UIScrollView here and the framework semantics tree having different opinion about what the content offset should be, the update takes time to travel and the inconsistency can be observed. 

If the communication is synchronous then the inconsistent state won't be observable as the updates would arrive instantaneously.